// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

syntax="proto2";
import "brpc/options.proto";

package brpc;

option java_package="com.brpc";
option java_outer_classname="NsheadProto";

message NsheadMeta {
    // Returned by `MethodDescriptor::full_name'
    // Something likd "test.EchoService.Echo"
    required string full_method_name = 1;      
    
    optional int64 correlation_id = 2;
    optional int64 log_id = 3;
    optional int32 attachment_size = 4;
    optional CompressType compress_type = 5;

    optional int64 trace_id = 6;
    optional int64 span_id = 7;
    optional int64 parent_span_id = 8;

    optional bytes user_string = 9;
}
